src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java

```diff
@@ -181,20 +226,19 @@ private void unpack(Artifact artifact) {
     return selectors;
   }
 
-  private ArtifactResult resolve(ArtifactRequest request) {
-    try {
-      return repository.resolveArtifact(session, request);
-    }
-    catch (ArtifactResolutionException e) {
-      throw new RuntimeException(e);
+  private Version highestVersion(Artifact artifact) {
+    if (assumeSnapshotsAreLocal()) {
+      if (artifact.getVersion().endsWith("-SNAPSHOT"))
+        return new VersionImplementation().withVersion(artifact);
+
+      if (artifact.getVersion().endsWith("-SNAPSHOT]") && artifact.getVersion().startsWith("["))
```

Comment:
If I'm reading this right you're assuming a ] doesn't come after the [ - that might not be healthy to do?

Reply:
its just checking to see if the version is [...-SNAPSHOT] i.e. a fixed snapshot version